pkg/cvo/metrics.go

```diff
@@ -346,8 +356,27 @@ func (m *operatorMetrics) Describe(ch chan<- *prometheus.Desc) {
 	ch <- m.clusterOperatorConditionTransitions.WithLabelValues("", "").Desc()
 	ch <- m.clusterInstaller.WithLabelValues("", "", "").Desc()
 	ch <- m.clusterVersionOperatorUpdateRetrievalTimestampSeconds.WithLabelValues("").Desc()
+	ch <- m.clusterVersionConditionalUpdatesRecommendedConditions.WithLabelValues("", "", "").Desc()
 }
 
+func (m *operatorMetrics) collectConditionalUpdates(ch chan<- prometheus.Metric, updates []configv1.ConditionalUpdate) {
+	for _, update := range updates {
+		for _, condition := range update.Conditions {
+			if condition.Type != ConditionalUpdateConditionTypeRecommended {
+				continue
+			}
+
+			g := m.clusterVersionConditionalUpdatesRecommendedConditions
+			gauge := g.WithLabelValues(update.Release.Version, strings.ToLower(string(condition.Status)), condition.Reason)
+			gauge.Set(m.nowFunc().Sub(condition.LastTransitionTime.Time).Seconds())
```

Comment:
It might be slightly more robust to serve the timestamp here (like [we do today for `cluster_version_operator_update_retrieval_timestamp_seconds`][1]) and then do the time subtraction [in the alert][2].  That way, we trip the alert after some failed scrapes, and we can report a to-the-second duration of staleness in the alert description, instead of being limited to to-the-last-scrape precision.  But if we start failing scrapes, there will be other alerts going off (`TargetDown`, possibly others), so not a big deal either way.

[1]: https://github.com/openshift/cluster-version-operator/blob/ddbf138b75f85bb880ec377bae389d060291298a/pkg/cvo/metrics.go#L540-L541
[2]: https://github.com/openshift/cluster-version-operator/blob/ddbf138b75f85bb880ec377bae389d060291298a/install/0000_90_cluster-version-operator_02_servicemonitor.yaml#L61-L62

Reply:
As a user I dislike these "timestamp when something happened" values in metrics as it results in very high integer values that do not move. They are hard to eyeball when exploring prometheus data and behave poorly when put in chart. It's a tradeoff :shrug: 

I have slight preference for keeping the metric the way I wrote it - I think the robustness you describe has value mostly for the critical alerts when "we lost data" is a reason to annoy the admin. When monitoring stack is in trouble then firing our alert likley just increases noise?